src/code/z_play.c

```diff
@@ -1,81 +1,1608 @@
+
+        Actor_DrawAll(this, &this->actorCtx);
+
+        if (!this->envCtx.sunMoonDisabled) {
+            sp25C.x = this->view.eye.x + this->envCtx.unk_4;
+            sp25C.y = this->view.eye.y + this->envCtx.unk_8;
+            sp25C.z = this->view.eye.z + this->envCtx.unk_C;
+            Environment_DrawSunLensFlare(this, &this->envCtx, &this->view, gfxCtx, sp25C);
+        }
+
+        Environment_DrawCustomLensFlare(this);
+
+        if (R_PLAY_FILL_SCREEN_ON) {
+            Environment_FillScreen(gfxCtx, R_PLAY_FILL_SCREEN_R, R_PLAY_FILL_SCREEN_G, R_PLAY_FILL_SCREEN_B,
+                                   R_PLAY_FILL_SCREEN_ALPHA, 3);
+        }
+
+        switch (this->envCtx.fillScreen) {
+            case 1:
+                Environment_FillScreen(gfxCtx, this->envCtx.screenFillColor[0], this->envCtx.screenFillColor[1],
+                                       this->envCtx.screenFillColor[2], this->envCtx.screenFillColor[3], 3);
+                break;
+            default:
+                break;
+        }
+
+        if (this->envCtx.sandstormState != 0) {
+            Environment_DrawSandstorm(this, this->envCtx.sandstormState);
+        }
+
+        if (this->unk_18876 != 0) {
+            Environment_FillScreen(gfxCtx, 0, 0, 0, this->unk_18876, 3);
+        }
+
+        DebugDisplay_DrawObjects(this);
+        Play_DrawMotionBlur(this);
+
+        if (((R_PAUSE_MENU_MODE == 1) || (gTrnsnUnkState == 1)) || (R_PICTOGRAPH_PHOTO_STATE == 1)) {
+            Gfx* sp74;
+            Gfx* sp70 = POLY_OPA_DISP;
+
+            sp74 = Graph_GfxPlusOne(sp70);
+            gSPDisplayList(OVERLAY_DISP++, sp74);
+            this->pauseBgPreRender.fbuf = gfxCtx->curFrameBuffer;
+
+            if (R_PAUSE_MENU_MODE == 1) {
+                R_PAUSE_MENU_MODE = 2;
+                this->pauseBgPreRender.fbufSave = (u16*)gfxCtx->zbuffer;
+                this->pauseBgPreRender.cvgSave = this->unk_18E58;
+            } else if (R_PICTOGRAPH_PHOTO_STATE == 1) {
+                R_PICTOGRAPH_PHOTO_STATE = 2;
+                this->pauseBgPreRender.fbufSave = (u16*)gfxCtx->zbuffer;
+                this->pauseBgPreRender.cvgSave = this->unk_18E58;
+            } else {
+                gTrnsnUnkState = 2;
+                this->pauseBgPreRender.fbufSave = (u16*)gfxCtx->zbuffer;
+                this->pauseBgPreRender.cvgSave = NULL;
+            }
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_play/func_80167DE4.s")
+            func_801705B4(&this->pauseBgPreRender, &sp74);
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_play/func_80167F0C.s")
+            if (this->pauseBgPreRender.cvgSave != NULL) {
+                func_80170730(&this->pauseBgPreRender, &sp74);
+            }
 
+            gSPEndDisplayList(sp74++);
+            Graph_BranchDlist(sp70, sp74);
+            POLY_OPA_DISP = sp74;
+            this->unk_18B49 = 2;
+            SREG(33) |= 1;
+        } else {
+        Play_Draw_PostWorldDraw:
+            Play_PostWorldDraw(this);
```

Comment:
the main point of this grouping of tasks in this function here is that they are things that should NOT appear in the screenshot taken by the pause menu for the background. 

oot calls this "overlay elements" currently, which is not the best name, but more gets the point across that its the hud, textboxes and other things that should not be drawn in the screenshot frame of the pause menu (or oots game over screen)

we can possibly come up with a good term for this and use it in both games. I dont think post world draw conveys that purpose as well as it can. but i dont have any ideas for a new name yet

Reply:
That seems more like a side effect than its intended purpose, especially since some of the interface does still draw when kaleido's open. The point for me was that neither the light adjustment nor the screen shrinking effect is really an overlay, and the distinguishing feature seemed to be that it drew after everything else.

Reply:
it is 100% its purpose, not a side effect. They draw in the pause menu yes, but the point is the SCREENSHOT image that is in the background. All of the things contained within this function are skipped during the prerender screenshot step so that it wont be in the picture. screen shrinking effect included.

Reply:
agree on overlay not being the correct term, but this is particular grouping of these drawing elements is soley for the purpose of the prerender image and should have a better name to reflect that. this is all controlled by 
`R_PAUSE_MENU_MODE` as it runs through its different states while taking the screenshot and processing the image

Reply:
also just noticed it skips this stuff for taking a pictograph photo, so the hud etc isnt in the picture. same concept though